daemon/endpoint.go

```diff
@@ -182,33 +172,50 @@ func (h *putEndpointID) Handle(params PutEndpointIDParams) middleware.Responder
 	}
 	ep.Mutex.Unlock()
 	if build {
-		if err := ep.RegenerateWait(h.d, reason); err != nil {
+		if err := ep.RegenerateWait(d, reason); err != nil {
 			ep.RemoveDirectory()
-			return apierror.Error(PatchEndpointIDFailedCode, err)
+			return PutEndpointIDFailedCode, err
 		}
 	}
 
 	ep.Mutex.RLock()
 	endpointmanager.Insert(ep)
 	ep.Mutex.RUnlock()
 
-	add := labels.NewLabelsFromModel(params.Endpoint.Labels)
+	add := labels.NewLabelsFromModel(lbls)
 
 	if len(add) > 0 {
-		errLabelsAdd := h.d.UpdateSecLabels(params.ID, add, labels.Labels{})
+		code, errLabelsAdd := d.UpdateSecLabels(id, add, labels.Labels{})
 		if errLabelsAdd != nil {
 			// XXX: Why should the endpoint remain in this case?
 			log.WithFields(logrus.Fields{
-				logfields.EndpointID:              params.ID,
+				logfields.EndpointID:              id,
 				logfields.IdentityLabels:          logfields.Repr(add),
 				logfields.IdentityLabels + ".bad": errLabelsAdd,
 			}).Error("Could not add labels while creating an ep due to bad labels")
-			return errLabelsAdd
+			return code, errLabelsAdd
 		}
 	}
 
-	ret := NewPutEndpointIDCreated()
-	return ret
+	return PutEndpointIDCreatedCode, nil
+}
+
+func (h *putEndpointID) Handle(params PutEndpointIDParams) middleware.Responder {
+	log.WithField(logfields.Params, logfields.Repr(params)).Debug("PUT /endpoint/{id} request")
+
+	epTemplate := params.Endpoint
+	if n, err := endpoint.ParseCiliumID(params.ID); err != nil {
+		return apierror.Error(PutEndpointIDInvalidCode, err)
+	} else if n != epTemplate.ID {
+		return apierror.New(PutEndpointIDInvalidCode,
+			"ID parameter does not match ID in endpoint parameter")
+	} else if epTemplate.ID == 0 {
+		return apierror.New(PutEndpointIDInvalidCode,
+			"endpoint ID cannot be 0")
+	}
+
+	code, err := h.d.createEndpoint(epTemplate, params.ID, params.Endpoint.Labels)
+	return apierror.Error(code, err)
```

Comment:
can't this be:
```
code, err := ...
if err != nil {
return apierror.Error(code, err)
}
ret := NewPutEndpointIDCreated()
return ret
```

Reply:
Yeah, I suppose that's a bit more clear.